### PR TITLE
Support adapt Log frameworks

### PR DESCRIPTION
**Pull request by mrproliu**
Support adapting `logrus` and `zap` log frameworks with the following parts:
1. Integrating Agent logs into the Service
2. Integrating Tracing information into the Service

### CODE REVIEW

File: docs/en/advanced-features/logging-setup.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+
+Agent logs output the current running status of the Agent system, most of which are execution exceptions. 
+For example, communication anomalies between the Agent and the backend service, plugin execution exceptions, etc.
+
+Integrating Agent logs into the service's logging system can effectively help users quickly troubleshoot whether there are issues with the current Agent execution.
+
+### Tracing information into the Service
+
+The Agent would also enhance the existing logging system. 
+When the service outputs log, if the current goroutine contains Tracing data, it would be outputted together with the current logs.
```
Comment (wu-sheng) [inline docs/en/advanced-features/logging-setup.md:25]: This reminds me, whether go agent has added the mechanism that would NOT generate any trace information if the backend is not connected.
This is important in the product env. Once the backend(OAP or any others) is down, there is no point to keep tracing.

File: docs/en/advanced-features/logging-setup.md [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+1. **TraceID**: The current Trace ID. If there is no link, it outputs `Noop`.
+2. **SegmentID**: The Segment ID in the current Trace. If there is no link, it outputs `Noop`.
+3. **SpanID**: The Span ID currently being operated on. If there is no link, it outputs `0`.
+
+The output format is as follows: `[${TraceID},${SegmentID},${SpanID}]`.
+
+The following is an example of a log output when using `Zap.NewProduction`:
+
+```
+{"level":"info","ts":1683635924.717511,"caller":"gin/main.go:45","msg":"test log","SW_CTX":"[6fbe5844ee6611ed8e30acde48001122,6fbe5902ee6611ed8e30acde48001122,0]"}
```
Comment (wu-sheng) [inline docs/en/advanced-features/logging-setup.md:41]: Is this the same format as Java agent's SW_CTX? It is better we are always consistent.

File: docs/en/advanced-features/logging-setup.md (hunk 1 above)
Comment (mrproliu) [reply docs/en/advanced-features/logging-setup.md:25]: Nope. For now, the gRPC will keep tracing if the connection is closed. I could create other PR for this.

File: docs/en/advanced-features/logging-setup.md (hunk 2 above)
Comment (mrproliu) [reply docs/en/advanced-features/logging-setup.md]: The format has been updated to the same as the Java agent.

**Review by wu-sheng — APPROVED** [review]
LGTM

**Pull request merged**